build.gradle

```diff
@@ -343,7 +345,7 @@ distributions {
                 exclude '**/build/**', 'doc_src/**/pdf/**', 'local.properties'
             }
             into('lib/provided') {
-                from configurations.runtime
+                from configurations.runtimeOnly
```

Comment:
Did you test that this actually results in all of the dependencies we need being copied? Because I think `runtimeOnly` is not equivalent to `runtime`. I think we need to include `compileOnly` as well.

Reply:
runtimeOnly + compileOnly = Implementation.

I'd like to use implementation here.

Reply:
Here is a place for source distribution. I may need to test whether  source distribution  compile well.

Reply:
Thanks for looking into it. Ideally we should test that the source distro compiles as part of our CI.